### PR TITLE
[SPARK-19236][SQL][FOLLOW-UP] Added createOrReplaceGlobalTempView method

### PR DESCRIPTION
**Pull request by gatorsmile**
### What changes were proposed in this pull request?
This PR does the following tasks:
- Added @ since 
- Added the Python API
- Added test cases 

### How was this patch tested?
Added test cases to both Scala and Python

### CODE REVIEW

File: python/pyspark/sql/dataframe.py [hunk 1]
```diff
@@ -191,6 +191,23 @@ def createGlobalTempView(self, name):
...
+    @since(2.2)
+    def createOrReplaceGlobalTempView(self, name):
+        """Creates or replaces a global temporary view using the given name.
+
+        The lifetime of this temporary view is tied to this Spark application.
+
+        >>> df.createOrReplaceGlobalTempView("people")
+        >>> df2 = df.filter(df.age > 3)
+        >>> df2.createOrReplaceGlobalTempView("people")
+        >>> df3 = spark.sql("select * from people")
```
Comment (dongjoon-hyun) [inline python/pyspark/sql/dataframe.py]: `df3 = spark.sql("select * from _global_temp.people")`?

File: sql/core/src/test/scala/org/apache/spark/sql/execution/GlobalTempViewSuite.scala [hunk 2]
```diff
@@ -35,39 +35,47 @@ class GlobalTempViewSuite extends QueryTest with SharedSQLContext {
...
-    sql(s"ALTER VIEW $globalTempDB.src2 AS SELECT 2, 'b'")
-    checkAnswer(spark.table(s"$globalTempDB.src2"), Row(2, "b"))
+      // We can also use Catalog API to drop global temp view
+      spark.catalog.dropGlobalTempView("src2")
+      intercept[NoSuchTableException](spark.table(s"$globalTempDB.src2"))
 
-    // We can also use Catalog API to drop global temp view
-    spark.catalog.dropGlobalTempView("src2")
-    intercept[NoSuchTableException](spark.table(s"$globalTempDB.src2"))
+      // We can also use Dataset API to replace global temp view
```
Comment (gatorsmile) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/GlobalTempViewSuite.scala]: The code added are line 38 and line 73-78

**Comment by gatorsmile** [comment]
cc @cloud-fan @ueshin

File: python/pyspark/sql/dataframe.py [hunk 3]
```diff
@@ -191,6 +191,23 @@ def createGlobalTempView(self, name):
         """
         self._jdf.createGlobalTempView(name)
 
+    @since(2.2)
```
Comment (cloud-fan) [inline python/pyspark/sql/dataframe.py]: we didn't merge the original PR to 2.2 right?

**Comment by ueshin** [comment]
LGTM except for @cloud-fan's comment.

File: python/pyspark/sql/dataframe.py (hunk 3 above)
Comment (gatorsmile) [inline python/pyspark/sql/dataframe.py]: See the backport PR https://github.com/apache/spark/pull/18167

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Pull request closed without merging**